graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/IndexTemplateAdapter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.storage.opensearch2;
+
+import org.graylog2.indexer.indices.Template;
+
+public interface IndexTemplateAdapter {
```

Comment:
This interface has no OS/ES specific code, so it can be moved somewhere else and does not need to be copied in elasticsearch package.

Some parts of duplicated code (i.e. `serialize...` methods) can be moved to this class than, to avoid duplication.